usr/lib/common/loadsave.c

```diff
@@ -1772,13 +1773,14 @@ static CK_RV aes_256_wrap(unsigned char out[40],
     EVP_CIPHER_CTX_set_flags(ctx, EVP_CIPHER_CTX_FLAG_WRAP_ALLOW);
 
     if (EVP_CipherInit_ex(ctx, EVP_aes_256_wrap(), NULL, kek, NULL, 1) != 1
-        || EVP_CipherUpdate(ctx, out, &outlen, in, 32) != 1
-        || EVP_CipherFinal_ex(ctx, out, &outlen) != 1) {
+        || EVP_CipherUpdate(ctx, buffer, &outlen, in, 32) != 1
+        || EVP_CipherFinal_ex(ctx, buffer, &outlen) != 1) {
```

Comment:
I think for Final it should be `buffer + outlen`, since Update may already have produced some output, and Final should append to the already produced output, instead of overwriting it.

Reply:
Verified this and changed the code.